analyzers/tests/SonarAnalyzer.Test/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocation.cs

```diff
@@ -47,6 +46,313 @@ public void InstanceReference_SetsNotNull_VB()
+
+                    collection = Untracked()
+                    Tag("Before3", collection)
+                    Dim i3 = Enumerable.{{method}}(collection)           ' Call as normal static method
+                    Tag("After3", collection)
+
+                    collection = Untracked()
+                    Tag("Before4", collection)
+                    Dim i4 = Enumerable.{{method.ToLower()}}(collection) ' Call as normal static method case insensitive
+                    Tag("After4", collection)
+
+                    collection = Untracked()
+                    Tag("Before5", collection)
+                    Dim i5 = collection.NonLinqExtensionNonNullChecking()
+                    Tag("After5", collection)
+
+                    collection = Untracked()
+                    Tag("Before6", collection)
+                    Dim i6 = collection.NonLinqExtensionNullChecking()
+                    Tag("After6", collection)
+                End Sub
+
+                Private Function Untracked() As IEnumerable(Of Integer)
+                    Return New List(Of Integer) From {1, 2, 3}
+                End Function
+
+                Private Shared Sub Tag(name As String, arg As Object)
+                End Sub
+            End Class
+
+            Public Module MyExtensions
+                <Runtime.CompilerServices.Extension>
+                Public Function NonLinqExtensionNonNullChecking(source As IEnumerable(Of Integer)) As Integer
+                    Return 42
+                End Function
+                <Runtime.CompilerServices.Extension>
+                Public Function NonLinqExtensionNullChecking(source As IEnumerable(Of Integer)) As Integer
+                    If source Is Nothing Then Throw New ArgumentNullException(NameOf(source))
+                    Return 42
+                End Function
+            End Module
+            """;
+        var validator = new SETestContext(code, AnalyzerLanguage.VisualBasic, Array.Empty<SymbolicCheck>()).Validator;
+        validator.ValidateContainsOperation(OperationKind.Invocation);
+        validator.TagValue("Before1").Should().BeNull();
+        validator.TagValue("After1").Should().HaveOnlyConstraint(ObjectConstraint.NotNull);
+        validator.TagValue("Before2").Should().BeNull();
+        validator.TagValue("After2").Should().HaveOnlyConstraint(ObjectConstraint.NotNull);
+        validator.TagValue("Before3").Should().BeNull();
+        validator.TagValue("After3").Should().HaveOnlyConstraint(ObjectConstraint.NotNull);
+        validator.TagValue("Before4").Should().BeNull();
+        validator.TagValue("After4").Should().HaveOnlyConstraint(ObjectConstraint.NotNull);
+        validator.TagValue("Before5").Should().BeNull();
+        validator.TagValue("After5").Should().BeNull();
+        validator.TagValue("Before6").Should().BeNull();
+        validator.TagValue("After6").Should().BeNull();
+    }
+
+    [DataTestMethod]
+    [DataRow("Count")]
```

Comment:
This test method checks that the Invocation processor does not set `NotNull` on the source of "fake" LINQ extensions, i.e. methods that looks like from LINQ, but they are not LINQ.
I have added two cases:
- one is a method that, if it were from LINQ, would do the null-check, that is `Count`
- one is a method that, if it were from LINQ, would not do the null-check, that is `AsEnumerable`
I haven't added more, because that cover the space of possibilities, and adding more methods would, in my opinion, unnecessary. Does that clarify?